database/vmaas_db_postgresql.sql

```diff
@@ -1307,34 +1307,61 @@ ON CONFLICT DO NOTHING;
 
 
 -- -----------------------------------------------------
--- Table vmaas.csaf_products
+-- Table vmaas.csaf_product
 -- -----------------------------------------------------
-CREATE TABLE IF NOT EXISTS csaf_products (
-  id             SERIAL,
-  cpe            TEXT NOT NULL, CHECK (NOT empty(cpe)),
-  package        TEXT NULL, CHECK (NOT empty(package)),
-  module         TEXT NULL, CHECK (NOT empty(module)),
-  CONSTRAINT unique_csaf_product UNIQUE (cpe, package, module),
-  PRIMARY KEY (id)
+CREATE TABLE IF NOT EXISTS csaf_product (
+  id                SERIAL,
+  cpe_id            INT NOT NULL,
+  package_name_id   INT NULL,
+  package_id        INT NULL,
+  module_stream_id  INT NULL,
+  PRIMARY KEY (id),
+  CONSTRAINT cpe_id
+    FOREIGN KEY (cpe_id)
+    REFERENCES cpe (id),
+  CONSTRAINT package_name_id
+    FOREIGN KEY (package_name_id)
+    REFERENCES package_name (id),
+  CONSTRAINT package_id
+    FOREIGN KEY (package_id)
+    REFERENCES package (id),
+  CONSTRAINT module_stream_id
+    FOREIGN KEY (module_stream_id)
+    REFERENCES module_stream (id),
```

Comment:
I'm starting to think against using FK of `module_stream` table. Problem is that from CSAF we get only the module name and module_stream_name, while the `module_stream` table contains different and more granular data. We're unable to match any row from `module_stream` precisely.

`module_stream` table data implies also:
- architecture
- stream version
- stream context

To have an idea how many different rows can be matched for a single `module`:`module_stream` pair:

```
select m.name, ms.stream_name, count(*) from module m join module_stream ms on m.id = ms.module_id group by m.name, ms.stream_name limit 10;
```

```
 NAME               | STREAM_NAME | COUNT
--------------------+-------------+-------
 php                | 8.2         | 12
 nginx              | 1.16        | 288
 rhn-tools          | 1.0         | 477
 scala              | 2.1         | 126
 log4j              | 2           | 90
 satellite-5-client | 1.0         | 357
 ruby               | 2.6         | 324
 postgresql         | 13          | 506
 perl-YAML          | 1.24        | 1284
 mariadb-devel      | 10.3        | 160
```

Reply:
We could either store the module information directly as `TEXT` or create some simplified module table with only unique `module:ms` string pairs.